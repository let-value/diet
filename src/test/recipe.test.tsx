import { test, expect } from "bun:test";

import {
	jsx,
	Recipe,
	Directions,
	Step,
	Ingredients,
	Ingredient,
	Preparation,
} from "@/scheme";

export const test_recipe = (
	<Recipe name="test recipe" meal="breakfast,lunch,dinner" servings="4">
		<Directions>
			<Step>step 1</Step>
			<Step>step 2</Step>
			<Step>step 3</Step>
		</Directions>
		<Ingredients>
			<Ingredient name="ingredient1" quantity="1g" category="category1" />
			<Ingredient category="category2">ingredient2 (raw)</Ingredient>
			<Ingredient key="ingredient3" quantity="20 slices" category="category3">
				Fresh ingredient 3
			</Ingredient>
		</Ingredients>
		<Preparation>
			<Step>
				take <Ingredient manipulation="boil">ingredient1</Ingredient> and boil
				it
			</Step>
			<Step>
				then take{" "}
				<Ingredient quantity="2kg" manipulation="fry">
					ingredient2
				</Ingredient>{" "}
				and fry in on a pan
			</Step>
			<Step duration={10}>
				after that take{" "}
				<Ingredient key="ingredient3" manipulation="boil">
					ingredient 3
				</Ingredient>{" "}
				and boil it for 10 minutes
			</Step>
		</Preparation>
	</Recipe>
);

test("jsx recipe", () => {
	expect(test_recipe).toMatchSnapshot();
});
