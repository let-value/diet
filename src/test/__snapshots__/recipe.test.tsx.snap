// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`jsx recipe 1`] = `
Recipe {
  "children": [
    Ingredients {
      "children": [
        Ingredient {
          "category": [
            "category1",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "ingredient1",
          "quantity": Unit {
            "dimensions": [
              1,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "fixPrefix": false,
            "skipAutomaticSimplification": true,
            "units": [
              {
                "power": 1,
                "prefix": {
                  "name": "",
                  "scientific": true,
                  "value": 1,
                },
                "unit": {
                  "base": {
                    "dimensions": [
                      1,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                    ],
                    "key": "MASS",
                  },
                  "dimensions": [
                    1,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                  ],
                  "name": "g",
                  "offset": 0,
                  "prefixes": {
                    "E": {
                      "name": "E",
                      "scientific": true,
                      "value": 1000000000000000000,
                    },
                    "G": {
                      "name": "G",
                      "scientific": true,
                      "value": 1000000000,
                    },
                    "M": {
                      "name": "M",
                      "scientific": true,
                      "value": 1000000,
                    },
                    "P": {
                      "name": "P",
                      "scientific": true,
                      "value": 1000000000000000,
                    },
                    "Q": {
                      "name": "Q",
                      "scientific": true,
                      "value": 1000000000000000000000000000000,
                    },
                    "R": {
                      "name": "R",
                      "scientific": true,
                      "value": 1000000000000000000000000000,
                    },
                    "T": {
                      "name": "T",
                      "scientific": true,
                      "value": 1000000000000,
                    },
                    "Y": {
                      "name": "Y",
                      "scientific": true,
                      "value": 1000000000000000000000000,
                    },
                    "Z": {
                      "name": "Z",
                      "scientific": true,
                      "value": 1000000000000000000000,
                    },
                    "a": {
                      "name": "a",
                      "scientific": true,
                      "value": 0.000000000000000001,
                    },
                    "c": {
                      "name": "c",
                      "scientific": false,
                      "value": 0.01,
                    },
                    "d": {
                      "name": "d",
                      "scientific": false,
                      "value": 0.1,
                    },
                    "da": {
                      "name": "da",
                      "scientific": false,
                      "value": 10,
                    },
                    "f": {
                      "name": "f",
                      "scientific": true,
                      "value": 0.000000000000001,
                    },
                    "h": {
                      "name": "h",
                      "scientific": false,
                      "value": 100,
                    },
                    "k": {
                      "name": "k",
                      "scientific": true,
                      "value": 1000,
                    },
                    "m": {
                      "name": "m",
                      "scientific": true,
                      "value": 0.001,
                    },
                    "n": {
                      "name": "n",
                      "scientific": true,
                      "value": 0.000000001,
                    },
                    "p": {
                      "name": "p",
                      "scientific": true,
                      "value": 0.000000000001,
                    },
                    "q": {
                      "name": "q",
                      "scientific": true,
                      "value": 0.000000000000000000000000000001,
                    },
                    "r": {
                      "name": "r",
                      "scientific": true,
                      "value": 0.000000000000000000000000001,
                    },
                    "u": {
                      "name": "u",
                      "scientific": true,
                      "value": 0.000001,
                    },
                    "y": {
                      "name": "y",
                      "scientific": true,
                      "value": 0.000000000000000000000001,
                    },
                    "z": {
                      "name": "z",
                      "scientific": true,
                      "value": 0.000000000000000000001,
                    },
                  },
                  "value": 0.001,
                },
              },
            ],
            "value": 0.001,
          },
        },
        Ingredient {
          "category": [
            "category2",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "ingredient2 (raw)",
          "quantity": undefined,
        },
        Ingredient {
          "category": [
            "category3",
          ],
          "description": undefined,
          "key": "ingredient3",
          "manipulation": [],
          "name": "Fresh ingredient 3",
          "quantity": Unit {
            "dimensions": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              1,
            ],
            "fixPrefix": false,
            "skipAutomaticSimplification": true,
            "units": [
              {
                "power": 1,
                "prefix": {
                  "name": "",
                  "scientific": true,
                  "value": 1,
                },
                "unit": {
                  "base": {
                    "dimensions": [
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      1,
                    ],
                    "key": "things_STUFF",
                  },
                  "dimensions": [
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                    1,
                  ],
                  "name": "slices",
                  "offset": 0,
                  "prefixes": {},
                  "value": 1,
                },
              },
            ],
            "value": 20,
          },
        },
        Ingredient {
          "category": [
            "category4",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "ingredient4",
          "quantity": Unit {
            "dimensions": [
              0,
              3,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "fixPrefix": false,
            "skipAutomaticSimplification": true,
            "units": [
              {
                "power": 1,
                "prefix": {
                  "name": "",
                  "scientific": true,
                  "value": 1,
                },
                "unit": {
                  "base": {
                    "dimensions": [
                      0,
                      3,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                    ],
                    "key": "VOLUME",
                  },
                  "dimensions": [
                    0,
                    3,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                  ],
                  "name": "tsp",
                  "offset": 0,
                  "prefixes": {},
                  "value": 0.000005,
                },
              },
            ],
            "value": 0.000005,
          },
        },
        Ingredient {
          "category": [
            "category5",
          ],
          "description": undefined,
          "key": undefined,
          "manipulation": [],
          "name": "ingredient5",
          "quantity": Unit {
            "dimensions": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "fixPrefix": false,
            "skipAutomaticSimplification": true,
            "units": [],
            "value": 1,
          },
        },
      ],
    },
    Preparation {
      "children": [
        Step {
          "children": [
            "take ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [
                "boil",
              ],
              "name": "ingredient1",
              "quantity": undefined,
            },
            " and boil it",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "then take",
            " ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": undefined,
              "manipulation": [
                "fry",
              ],
              "name": "ingredient2",
              "quantity": Unit {
                "dimensions": [
                  1,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                  0,
                ],
                "fixPrefix": false,
                "skipAutomaticSimplification": true,
                "units": [
                  {
                    "power": 1,
                    "prefix": {
                      "name": "k",
                      "scientific": true,
                      "value": 1000,
                    },
                    "unit": {
                      "base": {
                        "dimensions": [
                          1,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                          0,
                        ],
                        "key": "MASS",
                      },
                      "dimensions": [
                        1,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                      ],
                      "name": "g",
                      "offset": 0,
                      "prefixes": {
                        "E": {
                          "name": "E",
                          "scientific": true,
                          "value": 1000000000000000000,
                        },
                        "G": {
                          "name": "G",
                          "scientific": true,
                          "value": 1000000000,
                        },
                        "M": {
                          "name": "M",
                          "scientific": true,
                          "value": 1000000,
                        },
                        "P": {
                          "name": "P",
                          "scientific": true,
                          "value": 1000000000000000,
                        },
                        "Q": {
                          "name": "Q",
                          "scientific": true,
                          "value": 1000000000000000000000000000000,
                        },
                        "R": {
                          "name": "R",
                          "scientific": true,
                          "value": 1000000000000000000000000000,
                        },
                        "T": {
                          "name": "T",
                          "scientific": true,
                          "value": 1000000000000,
                        },
                        "Y": {
                          "name": "Y",
                          "scientific": true,
                          "value": 1000000000000000000000000,
                        },
                        "Z": {
                          "name": "Z",
                          "scientific": true,
                          "value": 1000000000000000000000,
                        },
                        "a": {
                          "name": "a",
                          "scientific": true,
                          "value": 0.000000000000000001,
                        },
                        "c": {
                          "name": "c",
                          "scientific": false,
                          "value": 0.01,
                        },
                        "d": {
                          "name": "d",
                          "scientific": false,
                          "value": 0.1,
                        },
                        "da": {
                          "name": "da",
                          "scientific": false,
                          "value": 10,
                        },
                        "f": {
                          "name": "f",
                          "scientific": true,
                          "value": 0.000000000000001,
                        },
                        "h": {
                          "name": "h",
                          "scientific": false,
                          "value": 100,
                        },
                        "k": {
                          "name": "k",
                          "scientific": true,
                          "value": 1000,
                        },
                        "m": {
                          "name": "m",
                          "scientific": true,
                          "value": 0.001,
                        },
                        "n": {
                          "name": "n",
                          "scientific": true,
                          "value": 0.000000001,
                        },
                        "p": {
                          "name": "p",
                          "scientific": true,
                          "value": 0.000000000001,
                        },
                        "q": {
                          "name": "q",
                          "scientific": true,
                          "value": 0.000000000000000000000000000001,
                        },
                        "r": {
                          "name": "r",
                          "scientific": true,
                          "value": 0.000000000000000000000000001,
                        },
                        "u": {
                          "name": "u",
                          "scientific": true,
                          "value": 0.000001,
                        },
                        "y": {
                          "name": "y",
                          "scientific": true,
                          "value": 0.000000000000000000000001,
                        },
                        "z": {
                          "name": "z",
                          "scientific": true,
                          "value": 0.000000000000000000001,
                        },
                      },
                      "value": 0.001,
                    },
                  },
                ],
                "value": 2,
              },
            },
            " ",
            "and fry in on a pan",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "after that take",
            " ",
            Ingredient {
              "category": [],
              "description": undefined,
              "key": "ingredient3",
              "manipulation": [
                "boil",
              ],
              "name": "ingredient 3",
              "quantity": undefined,
            },
            " ",
            "and boil it for 10 minutes",
          ],
          "duration": Unit {
            "dimensions": [
              0,
              0,
              1,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "fixPrefix": false,
            "skipAutomaticSimplification": true,
            "units": [
              {
                "power": 1,
                "prefix": {
                  "name": "",
                  "scientific": true,
                  "value": 1,
                },
                "unit": {
                  "base": {
                    "dimensions": [
                      0,
                      0,
                      1,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                    ],
                    "key": "TIME",
                  },
                  "dimensions": [
                    0,
                    0,
                    1,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                    0,
                  ],
                  "name": "minutes",
                  "offset": 0,
                  "prefixes": {},
                  "value": 60,
                },
              },
            ],
            "value": 600,
          },
        },
      ],
    },
    Directions {
      "children": [
        Step {
          "children": [
            "step 1",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "step 2",
          ],
          "duration": undefined,
        },
        Step {
          "children": [
            "step 3",
          ],
          "duration": undefined,
        },
      ],
    },
  ],
  "description": undefined,
  "meal": [
    "breakfast",
    "lunch",
    "dinner",
  ],
  "name": "test recipe",
  "servings": Unit {
    "dimensions": [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    "fixPrefix": false,
    "skipAutomaticSimplification": true,
    "units": [],
    "value": 4,
  },
}
`;
